modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1540,12 +1546,12 @@ private CompletableFuture<BinaryRow> resolveRowByPkForReadOnly(BinaryTuple pk, H
                 // This means that we can just resolve the state of this transaction.
                 checkWriteIntentsBelongSameTx(writeIntents);
 
-                return resolveTxState(
-                        new TablePartitionId(writeIntent.commitTableId(), writeIntent.commitPartitionId()),
-                        writeIntent.transactionId(),
-                        ts)
-                        .thenApply(readLastCommitted -> {
-                            if (readLastCommitted) {
+                return resolveTxState(writeIntent, ts)
+                        .thenApply(transactionReadable -> {
+                            // The value of `transactionReadable` is true - when transaction is committed and commit time <= read time.
+                            if (transactionReadable) {
+                                return findAny(writeIntents, wi -> !wi.isEmpty()).map(ReadResult::binaryRow).orElse(null);
```

Comment:
Here is your choice, but I prefer not to use the "else" clause when returning something from "if". Otherwise, the next block of code is moved to the right more than necessary.